### PR TITLE
[dev] update csi driver and operator versions to 2.14.0

### PR DESCRIPTION
**Pull request by yorchpave**
Since we are about to start building **master** CNSA builds (v5.2.3.0), I figured we need to bump the CSI driver and operator versions.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [X] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## How risky is this change?
- [X] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

File: driver/build/Dockerfile [hunk 1]
```diff
@@ -20,8 +20,8 @@ ENV REVISION $commit
 RUN CGO_ENABLED=0 GOOS=$TARGETOS GOARCH=$TARGETARCH go build -a -ldflags "-X 'main.gitCommit=${REVISION}' -extldflags '-static'" -o _output/ibm-spectrum-scale-csi ./cmd/ibm-spectrum-scale-csi
 # RUN chmod +x,u+s _output/ibm-spectrum-scale-csi
 
-FROM registry.access.redhat.com/ubi9-minimal:9.4-1227.1726694542
-ARG version=2.13.0
+FROM registry.access.redhat.com/ubi9-minimal:9.5
```
Comment (yorchpave) [inline driver/build/Dockerfile:23]: @Jainbrt @hemalathagajendran please note that I also updated the image tag so we start pulling `ubi9-minimal:9.5`.

File: generated/installer/ibm-spectrum-scale-csi-operator-ocp-rhel.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ metadata:
     product: ibm-spectrum-scale-csi
     release: ibm-spectrum-scale-csi-operator
   annotations:
-    productVersion: 2.13.0
+    productVersion: 2.14.0
 spec:
   replicas: 1
```
Comment (saurabhwani5) [inline generated/installer/ibm-spectrum-scale-csi-operator-ocp-rhel.yaml:15]: `generated/installer/ibm-spectrum-scale-csi-operator-ocp-rhel.yaml` is not required from CSI 2.13.0 , so we should remove this file

Comment (yorchpave) [reply generated/installer/ibm-spectrum-scale-csi-operator-ocp-rhel.yaml]: @saurabhwani5 thanks! I deleted the file 😄

**Review by Jainbrt — CHANGES_REQUESTED** [review]
`operator/build/Dockerfile` this file need update for 2.14.0 also.

**Comment by Jainbrt** [comment]
We should  also fix MakeFile (https://github.com/IBM/ibm-spectrum-scale-csi/blob/dev/operator/Makefile#L87) to remove the rhel-yaml file regeneration.

**Review by Jainbrt — APPROVED** [review]

**Review by hemalathagajendran — APPROVED** [review]

**Pull request merged**